etw/eventdata.go

```diff
@@ -0,0 +1,23 @@
+package etw
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// EventData maintains a buffer which builds up the data for an ETW event. It
+// needs to be paired with EventMetadata which describes the event.
+type EventData struct {
+	buffer bytes.Buffer
+}
+
+// NewEventData returns a new EventData with an empty buffer.
+func NewEventData() *EventData {
+	return &EventData{}

```

Comment:
NIT: If we don't initialize any fields I prefer the syntax `new(EventData)` but certainly not required.

Reply:
I mostly put this in for symmetry with how EventMetadata is initialized, and in case we need to do any work at init in the future. I'm not sure those are fantastic reasons, though. :)

Reply:
Eh, whatever. `new` is mostly historic and sync this is a struct we can use the `&` syntax. Just leave it.